sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/perPartitionCircuitBreaker/PerPartitionCircuitBreakerUtilities.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.perPartitionCircuitBreaker;
+
+import com.azure.cosmos.BridgeInternal;
+import com.azure.cosmos.CosmosDiagnostics;
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.RxDocumentServiceRequest;
+import com.azure.cosmos.implementation.Strings;
+import com.azure.cosmos.implementation.guava25.base.Preconditions;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class PerPartitionCircuitBreakerUtilities {
+
+    private static final Map<String, String> EMPTY_MAP = new HashMap<>();
+    private static final String BASE_EXCEPTION_MESSAGE = "FAILED IN Per-Partition Circuit Breaker: ";
+
+    public static void checkNotNull(
+        RxDocumentServiceRequest request,
+        Object paramValue,
+        String exceptionMessage) {
+
+        try {
+            Preconditions.checkNotNull(paramValue, exceptionMessage);
+        } catch (Exception e) {
```

Comment:
Instead of catching everything, can we just catch the exception that `Preconditions.checkNotNull` throws, maybe NPE?